apps/client/src/lib/recherche/queryContents.ts

```diff
@@ -63,24 +63,29 @@ export const getDefaultSortOption = (query: SearchQuery): SortOptions => {
  * @param secondaryThemes - boolean. Use the primary or secondary theme to filter.
  * @returns - list of dispositifs
  */
-const filterDispositifs = (
+export const filterDispositifs = (
   query: SearchQuery,
   dispositifs: GetDispositifsResponse[],
   secondaryThemes: boolean,
+  skip: FilterKey | undefined = undefined,
 ): GetDispositifsResponse[] => {
   const filterKeys = buildFilterKeys(query);
   const rule = getDisplayRule(query.type, filterKeys, query.sort);
 
   const filteredDispositifs = dispositifs
-    .filter((dispositif) => filterByThemeOrNeed(dispositif, query.themes, query.needs, secondaryThemes))
-    .filter((dispositif) => filterByLocations(dispositif, query.departments))
-    .filter((dispositif) => filterByAge(dispositif, query.age))
-    .filter((dispositif) => filterByFrenchLevel(dispositif, query.frenchLevel))
-    .filter((dispositif) => filterByLanguage(dispositif, query.language))
-    .filter((dispositif) => filterByPublic(dispositif, query.public))
-    .filter((dispositif) => filterByStatus(dispositif, query.status));
+    .filter(
+      (dispositif) => skip === "theme" || filterByThemeOrNeed(dispositif, query.themes, query.needs, secondaryThemes),
```

Comment:
Hummm, skip only sometimes but not always, it was a tricky one !
Just tested it, it looks good to me.

Reply:
Yes the issue was that to calculate the matching number of documents for each filter, you need to get the filtered documents *except* for the filter for which you are calculating the counts for !